### ISSUE TITLE
How to Improve FPS on Wemos D1 Mini

### ISSUE THREAD

**Issue by biejay** [scottlawsonbc/audio-reactive-led-strip#246]
Hey,
the software is awesome! I am using it on a macbook and send the processed data to a Wemos d1 mini, which works fine.
But i rarely get 60 FPS and sometimes it even drops to 20-30 FPS and the LED-Visualization really lags.
Is there a known bottle neck? 
I am thinking about these possible causes: 
- Wemos D1 Mini processes to slow (i have already set cpu to 160mhz)
- Wifi is unstable?
- MacBook processes to slow (but processing latency is always below 0.03ms)

So what is the best way to improve the FPS?
Thanks!

**Comment by bjornstevens** [comment]
try turning down the MIC_RATE, i accidently made this 441000 and cpu (on ryzen 2700x) ramped up to 100%. try something less than 44100 i would say.

**Comment by biejay** [comment]
> try turning down the MIC_RATE, i accidently made this 441000 and cpu (on ryzen 2700x) ramped up to 100%. try something less than 44100 i would say.

I don't think that this is a problem. I am not using the Mic because i re-linked the audio output. And the processing time is always pretty low and ist normally at about 25% CPU usage.

**Comment by bjornstevens** [comment]
Okay then i have no clue what it could be.

**Comment by joeybab3** [comment]
The wemos is not the issue if it is not receiving the FPS.
When you say 'FPS' are you referring to the visualization on your computer lagging and going down to 20-30fps or the visualization on the LED strip?
If the visualization is slow on your mac then there isn't much we can do about that except for trying the solutions above.

**Comment by biejay** [comment]
@joeybab3 it's both. The fps counter (60/60) goes down to like 30/60. The visualization in my Mac is smooth. But the LEDs are lagging/get stuck for some 200-300 ms and then the Animation flows again.
